src/analysis/lattice.h

```diff
@@ -19,32 +20,53 @@ constexpr bool has_getLeastUpperBound = std::
   is_invocable_r<void, decltype(&T::getLeastUpperBound), T, const T&>::value;
 template<typename T>
 constexpr bool has_isTop =
-  std::is_invocable_r<bool, decltype(T::isTop), const T&>::value;
+  std::is_invocable_r<bool, decltype(&T::isTop), T>::value;
 template<typename T>
 constexpr bool has_isBottom =
-  std::is_invocable_r<bool, decltype(T::isBottom), const T&>::value;
+  std::is_invocable_r<bool, decltype(&T::isBottom), T>::value;
 
 template<typename T>
 constexpr bool is_lattice =
   has_compare<T>&& has_getLeastUpperBound<T>&& has_isTop<T>&& has_isBottom<T>;
 
 // Represents a powerset lattice element (i.e. a set) as a bitvector. A true
 // means that an element is present in the set.
-template<size_t N> struct BitsetPowersetLattice {
-  std::bitset<N> value;
+class FinitePowersetLattice {
```

Comment:
Btw, it slightly confused me that this is "lattice" but really (as the comment explains) it is a "lattice *element*". But adding "element" to the name would probably not be worth it...

Reply:
Added lattice element class.